### PR TITLE
Add build docs script

### PR DESCRIPTION
**Pull request by ruflin**
Building the docs is now part of make testsuite

Building of the generators was also added to the testsuite.

### CODE REVIEW

**Comment by ruflin** [comment]
@dedemorton This means in the future you can just run `make docs-preview` instead of assembling the command yourself ...

**Comment by ruflin** [comment]
@andrewkroh I removed the doc build in Jenkins to replace it by this PR. Should we backport this to 1.3, at least the overall docs build?

File: Makefile [hunk 1]
```diff
@@ -3,11 +3,13 @@ COVERAGE_DIR=${BUILD_DIR}/coverage
...
 
-# Runs complete testsuites (unit, system, integration) for all beats,
-# with coverage and race detection.
+# Runs complete testsuites (unit, system, integration) for all beats with coverage and race detection.
+# Also it builds the docs and the generators
 .PHONY: testsuite
 testsuite:
 	$(foreach var,$(PROJECTS),$(MAKE) -C $(var) testsuite || exit 1;)
+	#$(MAKE) -C generate test
+	$(MAKE) docs
```
Comment (andrewkroh) [inline Makefile]: I would prefer to not have the docs build tied to testsuite. I would rather be able to specify if I want them both to run like `make testsuite docs`.

**Comment by andrewkroh** [comment]
Yeah, it would be good to backport at least the `make docs` part of this.

File: Makefile (hunk 1 above)
Comment (ruflin) [inline Makefile]: I kept it in to not have to change the jenkins command and it is currently incompatible with 1.3. but I will just update 1.3 very soon, so this is not a problem. I removed it.

**Pull request merged**